Updated requirements and initial demo guidelines (#75)
microsoft/ccf-app-samples: data-reconciliation-app/docs/data-schema-data-flow.md

## Generic Data Reconcilition App = Many Use Cases

We will build our application for use across any data schemas. Pending legal approvals, reference data may be a test case for our app.

Main idea: We build a generalized sample app for a bunch of institutions who want to collaborate on some set of data, and share results out on that data.

## Data Input Schema

Requirements:

- Members will agree upon their schema in advance - the only requirement is that this schema has a unique idenifier and attributes associated with the unique identifier.
- MVP: Our idenitifier and attributes types will be strings.
- MVP: We will only handle one attribute in this data reconciliation app. For example:

  | unique_id | attribute_1 |
  | --------- | ----------- |
  | A001      | XX          |
  | A002      | XY          |
  | A003      | XZ          |

- TODO: Brent to provide initial sample data set
- Stretch: Attribute types can be numerical as well.
- Stretch: Multiple attributes per unique id

## Data Schema Validation

> Our main audience for our sample app is a developer. A developer could take our sample app and extend it to their own use case. A developer could provide their input schema. Validation on the schema is not neccessary for our app, but it could be a simple check:

Reviewer: Nit:
```suggestion
Our main audience for our sample app is developers. A developer could take our sample app and extend it to their own use case. A developer could provide their input schema. Validation on the schema is not necessary for our app, but it could be a simple check:
```

## Data Reconiliation

> - Data is compared across all members. In all scenarios, one member is not the single source of truth and their data. In theory, all members' data carry equal weight to reach consensus.

Reviewer: ```suggestion
- Data is compared across all members carrying equal weight to reach the consensus. So, the source of truth doesn't rely on a single member or their data.
```

- Reconciliation is on each record, not on the entire data set.
- Please refer to [data-reconciliation.md](./data-reconciliation.md)

## Data Reporting

- API Endpoint: Members will query for results
  - Query by specific record (unqiue_id)
  - Query for all data
- In order for our app to provide a report on the data, a certain percantage of members need to have "voted" (or submited data) on a particular record. A `voting_threshold` may change based on the scenario and number of members. Therefore, we will make the `voting_threshold` configurable.
- For example, if `voting_threshold`=0.8,
  - Consortium of 5 members -> 4/5 members have to vote
  - Consortium of 3 members -> 2/3 members have to vote
- If a record is determined out of consensus with other members in the network, you cannot share the value of what other members had for that record.

## Data Results Schema
- MVP:
  - Results data is categorial - input data attribute value is a string, so status of attribute relative to network will be `in_consensus`, `lack_of_consensus`, or `not_enough_votes`.
  - Results data will be returned as a CSV table to each members. 
  - Based on the member, table will look like:

>     | unique_id | attribute_n                 | my_status                                                                                               | count_of_unique_values                               | members_in_agreement                                              | majority_minority                                                                     |

Reviewer 2: Change "my_status" to "group_status"
